src/cdk/v2/destinations/the_trade_desk/utils.js

```diff
@@ -0,0 +1,130 @@
+
+const getSignatureHeader = (request, secretKey) => {
+  if (!secretKey) {
+    throw new AbortedError('Secret key is missing. Aborting');
+  }
+  const sha1 = CryptoJS.HmacSHA1(JSON.stringify(request), secretKey);
+  const base = CryptoJS.enc.Base64.stringify(sha1);
+  return base;
+};
+
+const responseBuilder = (items, Config) => {
+  const { advertiserId, dataServer } = Config;
+
+  const payload = { DataProviderId: DATA_PROVIDER_ID, AdvertiserId: advertiserId, Items: items };
+
+  const response = defaultRequestConfig();
+  response.endpoint = getFirstPartyEndpoint(dataServer);
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return response;
+};
+
+const splitItemsBasedOnMaxSizeInBytes = (items, maxSize) => {
+  if (!items || items.length === 0) return [];
+
+  const itemsSize = jsonSize(items);
+
+  if (itemsSize <= maxSize) {
+    return [items];
+  }
+
+  const batches = [];
+  let currentBatch = [];
+
+  items.forEach((item) => {
+    const itemSize = jsonSize(item);
+
+    if (jsonSize(currentBatch) + itemSize <= maxSize) {
+      currentBatch.push(item);
+    } else {
+      batches.push([...currentBatch]);
+      currentBatch = [item];
+    }
+  });
+
+  if (currentBatch.length > 0) {
+    batches.push(currentBatch);
+  }
+
+  return batches;
+};
+
+const batchResponseBuilder = (items, Config) => {
+  const response = [];
+  const itemsChunks = splitItemsBasedOnMaxSizeInBytes(items, MAX_REQUEST_SIZE_IN_BYTES);
+
+  itemsChunks.forEach((chunk) => {
+    response.push(responseBuilder(chunk, Config));
+  });
+
+  return response;
+};
+
+const processRecordInputs = (inputs, destination) => {
+  const { Config } = destination;
+  const items = [];
+  const successMetadata = [];
+  const errorResponseList = [];
+
+  const error = new InstrumentationError('Invalid action type');
+
+  inputs.forEach((input) => {
+    const { fields, action } = input.message;
+    const isInsertOrDelete = action === 'insert' || action === 'delete';
+
+    if (isInsertOrDelete) {
+      successMetadata.push(input.metadata);
+      const data = [
+        {
+          Name: Config.audienceId,
+          TTLInMinutes: action === 'insert' ? ttlInMin(Config.ttlInDays) : 0,
+        },
+      ];
+      Object.keys(fields).forEach((id) => {
+        items.push({ [id]: fields[id], Data: data });
+      });
+    } else {
+      errorResponseList.push(handleRtTfSingleEventError(input, error, {}));
+    }
+  });
+
+  const payloads = batchResponseBuilder(items, Config);
+
+  const response = getSuccessRespEvents(payloads, successMetadata, destination, true);
+  return [response, ...errorResponseList];
+};
+
+const processRouterDest = (inputs) => {
+  const respList = [];
+  const { destination } = inputs[0];
+  const groupedInputs = lodash.groupBy(inputs, (input) => input.message.type);
+  if (groupedInputs.record) {
```

Comment:
are we expecting only record type events?

Reply:
yes